src/atoms/Button/Button.web.js

```diff
@@ -1,20 +1,454 @@
+          return 'small';
+        }
+        return 'medium';
+      case 'medium':
+        if (children) {
+          return 'medium';
+        }
+        return 'large';
+      case 'large':
+        if (children) {
+          return 'medium';
+        }
+        return 'large';
+      default:
+        if (children) {
+          return 'medium';
+        }
+        return 'large';
+    }
+  },
+  fontSize({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xxsmall';
+      case 'small':
+        return 'xsmall';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  align({ align }) {
+    switch (align) {
+      case 'left':
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+      default:
+        return 'flex-start';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconAlign }) {
+    switch (size) {
+      case 'xsmall':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'small':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'medium':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      case 'large':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      default:
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+};
+
+const StyledButton = styled.button`
+  &&& {
+    background-color: ${styles.backgroundColor};
+    border-radius: ${(props) => props.theme.spacings.xxsmall};
+    border: ${styles.border};
+    width: ${(props) => (props.block ? '100%' : '')};
+    outline: none;
+    &:hover {
+      border: ${styles.hoverBorder};
+      background-color: ${styles.hoverBackgroundColor};
+    }
+    &:focus {
+      border: ${styles.focusBorder};
+      background-color: ${styles.focusBackgroundColor};
+    }
+    &:active {
+      border: ${styles.activeBorder};
+      background-color: ${styles.activeBackgroundColor};
+    }
+  }
+`;
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconAlign,
+  align,
+  block,
+}) => {
+  const theme = useContext(ThemeContext);
+
+  const color = `${variantColor || 'primary'}`;
```

Comment:
What's the difference between `color` and `variantColor`? Getting confused.

Reply:
The consumer can use their own `variantColor` as well. It is a prop on button whereas color is just an alias for it being used in many places.

Reply:
I've removed all other instances of `variantColor` so it won;t be much confusing now.

Reply:
I see this `color` approach in native implementation as well. @shubhnik Any specific reason for doing this for native?

Reply:
Makes sense. Replaced it.